docusaurus/docs/how-to-guides/data-source-plugins/add-migration-handler-for-backend.md

```diff
@@ -0,0 +1,216 @@
+1. The plugin must be run on a Grafana version 11.4 or later.
+
+More information about these prerequisites is found in [step 5](#step-5-use-migration-code-from-the-frontend-using-experimental-apis) but if your plugin can't adhere to these requirements, there is an alternative approach using existing APIs
+
+## Implement a backend migration handler
+
+To implement a backend migration handler, add migration logic in your backend code. This guide walks you through the steps necessary to use the latest tooling to automatically migrate queries and avoid duplicating code.
+
+:::note
+
+The migration system detailed in this guide doesn't support two-way migrations. Only forward migrations are covered. Query migrations aren't automatically persisted, so users need to manually save changes to ensure the process works as expected.
+
+:::
+
+### Step 1 (optional): Add a query schema
+
+First of all, plugins don't need to have strongly typed queries. While this lowers the barrier for plugin development, plugins that don't define types are harder to scale and maintain. The first step in this guide is to add the required files to define the plugin query.
+
+See the following example: [grafana-plugin-examples#400](https://github.com/grafana/grafana-plugin-examples/pull/400). As you can see, there are multiple files to create. These files are used for both generating OpenAPI documentation and validating that the received queries are valid.
+
+Create these files:
+
+- `query.go`: This file defines the Golang types for your query. For automatic migrations to work, it's important that your query extends the new `v0alpha1.CommonQueryProperties`. After that, just define your query custom properties.
+- `query_test.go`: This test file is both used to check that all the JSON files are up to date with the query model and to generate them. The first time you execute the test, it will generate these files (so take into account that `query.types.json` needs to exist, even if it's empty).
+- `query.*.json`: Automatically generated files. These schemas can be used for OpenAPI documentation, but it's a feature still in progress that isn't available yet.
+
+### Step 2: Changing the query model
+
+:::note
+
+For a complete example of how to add a query migration (steps 2, 3 and 4), refer to the code for [experimental APIs](https://github.com/grafana/grafana-plugin-examples/pull/403/files) or [stable APIs](https://github.com/grafana/grafana-plugin-examples/pull/407).
+
+:::
+
+Once your plugin has its own schemas, add the model change. Since queries within the major version (or same API version) need to be compatible, maintain a reference to the legacy data format. This reference helps to enable an easy migration path.
```

Comment:
I think this paragraph is a bit hard to read, maybe something like this would be a bit easier to understand for the first time readers:

```suggestion
Once your plugin has its own schemas, you can start introducing model changes. Since queries within the major version (or same API version) need to be compatible, you have to maintain a reference to the legacy data format. This reference helps you to enable an easy migration path.
```